src/_context/SearchFiltersContext.tsx

```diff
@@ -0,0 +1,74 @@
+import { createContext, FC, PropsWithChildren, useContext, useMemo, useState } from 'react';
+
+import { TSearchFilters } from '../shop/_models/searchFilters';
+
+const initialFilters: TSearchFilters = {
+  includeAllCardSystems: false,
+};
+
+type TSearchFiltersContext = {
+  /** The number of filters that were applied by the user */
+  appliedFiltersCount: number;
+  /** Undoes all the changes the user made that were not persisted */
+  clearChangedFilters: () => void;
+  /** Retrieves the value of a given filter from the persisted state */
+  getFilter: <T extends keyof TSearchFilters>(key: T, options?: { includeChangedFilters?: boolean }) => TSearchFilters[T];
+  /** Persists the state so that the filters can be applied and used to filter data  */
+  persistChangedFilters: () => void;
+  /** Resets the filters to the initial set */
+  resetFilters: () => void;
+  /** Saves the filter into a temporary state. These will not be applied until you persist the state. */
+  setFilter: <T extends keyof TSearchFilters>(key: T, value: TSearchFilters[T]) => void;
+};
+
+export const SearchFilterContext = createContext<TSearchFiltersContext>({
```

Comment:
PP: context initialisen met null, dat draagt toch weinig bij, gezien je alles zet in de component zelf